### PR TITLE
Cadence 1.0 upgrades

### PR DESCRIPTION
**Pull request by joshuahannan**
## Description

Upgrades `USDCFlow` to Cadence 1.0 and adds a minter that conforms to the bridge interface. The minter can be sent to the bridge with a public function that takes the bridge address as an argument, creates a minter, and sends it.

Still haven't been able to fully test it yet because I haven't tried the bridge standup script, but I will soon

### CODE REVIEW

File: contracts/USDCFlow.cdc [hunk 1]
```diff
@@ -18,159 +20,207 @@ import "FiatToken"
...
-    pub fun createEmptyVault(): @Vault {
-        return <-create Vault(balance: 0.0)
-    }
+    access(all) resource Minter: FlowEVMBridgeHandlerInterfaces.TokenMinter {
 
-    /// wrapFiatToken
-    ///
-    /// Provides a way for users to exchange a FiatToken Vault
-    /// for a USDCFlow Vault with the same balance
-    pub fun wrapFiatToken(_ from: @FungibleToken.Vault): @Vault {
```
Comment (sisyphusSmiling) [inline contracts/USDCFlow.cdc:153]: Are we no longer allowing users to wrap FiatToken to USDCFlow come Cadence 1.0? I thought one of the main goals was to have a redemption path for existing FiatToken Vaults via wrapping their old vault, but I could have misunderstood the aim

File: contracts/USDCFlow.cdc [hunk 2]
```diff
@@ -18,159 +20,207 @@ import "FiatToken"
...
     }
 
-    /// createEmptyVault
-    ///
-    /// @return The new Vault resource with a balance of zero
-    ///
-    pub fun createEmptyVault(): @Vault {
-        return <-create Vault(balance: 0.0)
-    }
+    access(all) resource Minter: FlowEVMBridgeHandlerInterfaces.TokenMinter {
```
Comment (sisyphusSmiling) [inline contracts/USDCFlow.cdc:169]: I don't think it's an issue, but have we confirmed that adding the interface conformance in the 1.0 version migrates successfully?

File: contracts/USDCFlow.cdc [hunk 3]
```diff
@@ -18,159 +20,207 @@ import "FiatToken"
...
+        }
 
-        emit TokensMinted(amount: wrappedFiatTokenVault.balance, depositedUUID: vault.uuid, mintedUUID: wrappedFiatTokenVault.uuid)
+        /// Function for the bridge to burn tokens that are bridged back to Flow EVM
+        access(all) fun burn(vault: @{FungibleToken.Vault}) {
+            let toBurn <- vault as! @USDCFlow.Vault
+            let amount = toBurn.balance
 
-        fiatTokenVaultRef.deposit(from: <-vault)
+            assert(USDCFlow.totalSupply >= amount, message: "burning more than total supply")
```
Comment (sisyphusSmiling) [inline contracts/USDCFlow.cdc:192]: Should this be added to `burnCallback()`? If the check fails here I could just call `Burner.burn` passing this vault and bypass this check. If the amount is in fact >= totalSupply, we'd hit underflow there

File: contracts/USDCFlow.cdc (hunk 1 above)
Comment (joshuahannan) [reply contracts/USDCFlow.cdc:153]: That was never the plan. Circle is doing redemptions via a snapshot of balances at the time of the upgrade, so allowing wrapping after Crescendo would enable double spending.

File: contracts/USDCFlow.cdc (hunk 2 above)
Comment (joshuahannan) [reply contracts/USDCFlow.cdc:169]: We're adding an entirely new resource type here, so there shouldn't be an interface conformance issue with this

File: contracts/USDCFlow.cdc [hunk 4]
```diff
@@ -18,159 +20,207 @@ import "FiatToken"
...
         init(balance: UFix64) {
             self.balance = balance
         }
 
+        /// Called when a fungible token is burned via the `Burner.burn()` method
+        /// The total supply will only reflect the supply in the Cadence version
+        /// of the FiatToken smart contract
+        access(contract) fun burnCallback() {
+            if self.balance > 0.0 {
+                USDCFlow.totalSupply = USDCFlow.totalSupply - self.balance
```
Comment (sisyphusSmiling) [inline contracts/USDCFlow.cdc:106]: same comment as below about `.burn`

Also, do we want to emit `Burned` here when `balance > 0.0`?

File: contracts/USDCFlow.cdc [hunk 5]
```diff
@@ -18,159 +20,207 @@ import "FiatToken"
...
-    pub resource Vault: FungibleToken.Provider, FungibleToken.Receiver, FungibleToken.Balance, MetadataViews.Resolver {
+    access(all) fun resolveContractView(resourceType: Type?, viewType: Type): AnyStruct? {
+        switch viewType {
+            case Type<FungibleTokenMetadataViews.FTView>():
+                return FungibleTokenMetadataViews.FTView(
+                    ftDisplay: self.resolveContractView(resourceType: nil, viewType: Type<FungibleTokenMetadataViews.FTDisplay>()) as! FungibleTokenMetadataViews.FTDisplay?,
+                    ftVaultData: self.resolveContractView(resourceType: nil, viewType: Type<FungibleTokenMetadataViews.FTVaultData>()) as! FungibleTokenMetadataViews.FTVaultData?
+                )
+            case Type<FungibleTokenMetadataViews.FTDisplay>():
+                // Potential TODO: Replace with USDC http file and media type?
```
Comment (sisyphusSmiling) [inline contracts/USDCFlow.cdc:54]: Can this comment be removed?

File: contracts/USDCFlow.cdc (hunk 2 above)
Comment (sisyphusSmiling) [reply contracts/USDCFlow.cdc:169]: Ah right ofc

File: contracts/USDCFlow.cdc (hunk 1 above)
Comment (sisyphusSmiling) [reply contracts/USDCFlow.cdc:153]: Got it, thanks for clarifying

**Review by sisyphusSmiling — COMMENTED** [review]
Final couple questions about burning

File: contracts/USDCFlow.cdc (hunk 4 above)
Comment (joshuahannan) [reply contracts/USDCFlow.cdc:106]: both good suggestions! I added them to `burnCallback`

**Review by sisyphusSmiling — APPROVED** [review]

**Pull request merged**